relay/server.go

```diff
@@ -306,43 +307,43 @@ func (s *Server) GetChunks(ctx context.Context, request *pb.GetChunksRequest) (*
 	clientID := string(request.OperatorId)
 	err := s.chunkRateLimiter.BeginGetChunkOperation(time.Now(), clientID)
 	if err != nil {
-		return nil, err
+		return nil, api.NewErrorResourceExhausted(fmt.Sprintf("rate limit exceeded: %v", err))
 	}
 	defer s.chunkRateLimiter.FinishGetChunkOperation(clientID)
 
 	// keys might contain duplicate keys
 	keys, err := getKeysFromChunkRequest(request)
 	if err != nil {
-		return nil, err
+		return nil, api.NewErrorInvalidArg(fmt.Sprintf("invalid request: %v", err))
 	}
 
 	mMap, err := s.metadataProvider.GetMetadataForBlobs(ctx, keys)
 	if err != nil {
-		return nil, fmt.Errorf(
-			"error fetching metadata for blob, check if blob exists and is assigned to this relay: %w", err)
+		return nil, api.NewErrorInternal(fmt.Sprintf(
+			"error fetching metadata for blob, check if blob exists and is assigned to this relay: %v", err))
 	}
 
 	finishedFetchingMetadata := time.Now()
 	s.metrics.ReportChunkMetadataLatency(finishedFetchingMetadata.Sub(finishedAuthenticating))
 
 	requiredBandwidth, err := computeChunkRequestRequiredBandwidth(request, mMap)
 	if err != nil {
-		return nil, fmt.Errorf("error computing required bandwidth: %w", err)
+		return nil, api.NewErrorInvalidArg(fmt.Sprintf("error computing required bandwidth: %v", err))
 	}
 	err = s.chunkRateLimiter.RequestGetChunkBandwidth(time.Now(), clientID, requiredBandwidth)
 	if err != nil {
```

Comment:
Is it possible that rate limiter failed (not that there is no budget left)?

Reply:
There is one case where it can be internal, will add that.
```
	limiter, ok := l.perClientBandwidthLimiter[requesterID]
	if !ok {
		return fmt.Errorf("internal error, unable to find bandwidth limiter for client ID %s", requesterID)
	}
```